src/main/scala/cromwell/binding/package.scala

```diff
@@ -30,4 +34,12 @@ package object binding {
     def serialize: Option[String] = attribute("serialize")
     def attribute(attr: String): Option[String] = runtimeAttributes.get(attr)
   }
+
+  /**
+   * Core data identifying a workflow including its unique ID, its namespace, and strongly typed inputs.
+   */
+  case class WorkflowDescriptor(namespace: WdlNamespace, actualInputs: WorkflowCoercedInputs) {
+    val id = UUID.randomUUID()
```

Comment:
I wanted to free callers from having to generate the UUID when we currently don't require that.  Once we support restart that may change.